lib/spack/spack/spec.py

```diff
@@ -970,7 +970,115 @@ def __init__(self, spec, name, query_parameters):
 @key_ordering
 class Spec(object):
 
-    def __init__(self, spec_like, *dep_like, **kwargs):
+    @staticmethod
+    def from_literal(spec_dict, unique=True):
```

Comment:
That's the only point I didn't correct. 

Right now we have `Spec.from_yaml`, `Spec.from_json` and `Spec.from_dict` (the former two being implemented in terms of the latter). Other 6 classes related to `Spec` (like `ArchSpec`, etc.) implement a `from_dict` method, which is used as part of `Spec.from_node_dict`. 

Should I turn `Spec.from_dict` into `Spec._from_json_yaml_dict` so that I can reserve the name for the current `Spec.from_literal`? Should we care in this PR of naming consistency for the other classes?